### PR TITLE
Host validations- improved error messages, added sudo check

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Improved the error messages for host validations
Improved the gather facts efficiency by removing loop

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/confluent.common/tasks/host_validations.yml [hunk 1]
```diff
@@ -1,22 +1,37 @@
 ---
+- name: Confirm Running as Root
+  assert:
+    that: ansible_user_id == 'root'
+    fail_msg: >-
+      CP-Ansible must have sudo access. Current User ID: {{ansible_user_id}} is not root.
+      Make sure to set ansible_become to true and provide sudo access.
+      To skip host validations, set validate_hosts to false.
+  tags: validate
```
Comment (domenicbove) [inline roles/confluent.common/tasks/host_validations.yml:9]: @ldom Added you as a reviewer bc this task will need a conditional in your branch

File: roles/confluent.common/tasks/host_validations.yml [hunk 2]
```diff
@@ -1,22 +1,37 @@
 ---
+- name: Confirm Running as Root
+  assert:
+    that: ansible_user_id == 'root'
```
[the remark this answers is not in the document]
Comment (domenicbove) [reply roles/confluent.common/tasks/host_validations.yml:4]: So the intention of this task is not to check if `ansible_become == true` and `ansible_become_user == root` but to test if ansible could successfully "become" (sudo to) root. Some users have locked down envs where the ssh user does not have sudo access and even setting those vars correctly still doesnt work. I can change the error message

[the remark this answers is not in the document]
Comment (domenicbove) [reply roles/confluent.common/tasks/host_validations.yml:4]: actually after more testing... the playbook will error out before getting to this assertion if it cant sudo properly
```
TASK [confluent.common : Gather OS Facts] **************************************
fatal: [ksql1]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/bin/python"}, "changed": false, "module_stderr": "sudo: a password is required\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

Maybe this assertion is overkill...

File: roles/confluent.common/tasks/host_validations.yml [hunk 3]
```diff
@@ -1,22 +1,37 @@
...
+      Make sure to set ansible_become to true and provide sudo access.
+      To skip host validations, set validate_hosts to false.
+  tags: validate
+
 - name: Confirm Rhel Version Supported
   assert:
     that: ansible_distribution_major_version in ['7', '8']
-    fail_msg: "Rhel Version: {{ansible_distribution_major_version}} not in supported versions: 7, 8"
+    fail_msg: >-
+      Rhel Version: {{ansible_distribution_major_version}} not in supported versions: 7, 8.
```
[the remark this answers is not in the document]
Comment (domenicbove) [reply roles/confluent.common/tasks/host_validations.yml]: added var

File: roles/confluent.common/tasks/host_validations.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (erikgb) [reply roles/confluent.common/tasks/host_validations.yml]: > Maybe this assertion is overkill...

I think so. The requirement is clearly documented and this is a classical Ansible "thing".

File: roles/confluent.common/vars/main.yml [hunk 4]
```diff
@@ -10,3 +10,7 @@ confluent_cli_goarch:
 
 ### Filename of Confluent CLI binary inside the archive
 confluent_cli_binary: confluent
+
+redhat_supported_versions: ['7', '8']
+ubuntu_supported_versions: ['16', '18']
+debian_supported_versions: ['9']
```
Comment (erikgb) [inline roles/confluent.common/vars/main.yml:16]: I was thinking about variables on the tasks. Sorry for not being precise. Role variables seems like a bit overkill...

Comment (domenicbove) [reply roles/confluent.common/vars/main.yml:16]: yep haha good thinking!

**Review by erikgb — APPROVED** [review]
LGTM

**Pull request merged**